cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -262,8 +262,19 @@ func ValidateCloudProvider(provider string, fldPath *field.Path) field.ErrorList
 }
 
 func ValidateMixedArguments(flag *pflag.FlagSet) error {
-	if flag.Changed("config") && flag.NFlag() != 1 {
-		return fmt.Errorf("can not mix '--config' with other arguments")
+	if flag.Changed("config") {
+		numFlagsAllowedToChange := 1
+		flagsAllowedToChangeWithConfig := []string{"skip-preflight-checks", "skip-token-print"}
```

Comment:
Can't you just check to see if the flags are anything other then these two and output an error indicating the flags they specified?